java/code/src/com/suse/manager/model/hub/HubFactory.java

```diff
@@ -361,4 +376,29 @@ public List<AccessTokenDTO> listAccessToken(int offset, int pageSize) {
             ))
             .toList();
     }
+
+    /**
+     * Return list of {@link ChannelInfoDetailsJson} for a given {@link IssPeripheral} server
+     * @param peripheral the peripheral server
+     * @return return a list of ChannelInfoDetails for synchronization with the peripheral
+     */
+    public List<ChannelInfoDetailsJson> listChannelInfoForPeripheral(IssPeripheral peripheral) {
+        List<ChannelInfoDetailsJson> result = new ArrayList<>();
+        List<IssPeripheralChannels> peripheralChannels = listIssPeripheralChannels(peripheral);
+        List<Channel> channels = peripheralChannels.stream().map(IssPeripheralChannels::getChannel).toList();
```

Comment:
Do we really need to use the full `Channel` entity here? Can't we just use the `id` or the `label` to perform the `contains()` of line 395? From what I see, the `Channel::equal` compares only the `id` anyway. 

Moreover, it would be better to use an `HashSet`, so we can get constant time `contains()` and improve performance:

```java
        Set<Long> channelIds = peripheralChannels.stream()
            .map(IssPeripheralChannels::getChannel)
            .map(Channel::getId)
            .collect(Collectors.toCollection(HashSet::new));
```